addons/web/static/tests/views/kanban_tests.js

```diff
@@ -1583,6 +1583,61 @@ QUnit.module('Views', {
         kanban.destroy();
     });
 
+    QUnit.test('quick create record: do not cancel on window click when many2one autocomplete is open', async function (assert) {
```

Comment:
This isn't the problem you have to solve, imho. I think what you do is not closing the kanban record if the user clicks outside but mousedown inside beforehand